crates/sargon/src/profile/v100/profile.rs

```diff
@@ -225,12 +225,66 @@ impl Profile {
     }
 }
 
+impl<T: IsEntity> IdentifiedVecOf<T> {
+    pub fn erased(&self) -> IdentifiedVecOf<AccountOrPersona> {
+        self.items()
+            .into_iter()
+            .map(Into::<AccountOrPersona>::into)
+            .collect()
+    }
+}
+
 impl Profile {
     /// Returns the unique ID of this Profile (just an alias for `header.id`).
     pub fn id(&self) -> ProfileID {
         self.header.id
     }
 
+    /// Checks ALL FactorInstances for ALL Personas and Accounts on ALL networks,
+    /// returns Err(CommonError::FactorInstancesDiscrepancy { .. }) if the same
+    /// FactorInstances is used between any entity.
+    pub fn assert_factor_instances_valid(&self) -> Result<()> {
```

Comment:
This uncovered the bug in Android host - where FactorInstances was shared between Personas and Accounts, which was fixed in https://github.com/radixdlt/babylon-wallet-android/pull/1256